src/textual/renderables/sparkline.py

```diff
@@ -73,13 +74,15 @@ def __rich_console__(
         minimum, maximum = min(self.data), max(self.data)
         extent = maximum - minimum or 1
 
-        buckets = list(self._buckets(self.data, num_buckets=self.width))
+        buckets = tuple(self._buckets(self.data, num_buckets=width))
 
         bucket_index = 0
         bars_rendered = 0
-        step = len(buckets) / width
+        step = len(buckets) // width
```

Comment:
**NOTE** -- I *believe* this is the intention, but not sure. Either way this failed strict typing, which is how I noticed.